### PR TITLE
[FD-1384] ✨ Add Search API Support to Terminologies

### PR DESCRIPTION
**Pull request by brendagutman**

### CODE REVIEW

File: src/locutus/model/terminology.py [hunk 1]
```diff
@@ -44,12 +44,44 @@ def message(self):
...
+            Args:
+                data (dict): The deserialized data.
+            Returns:
+                OntologyAPISearchPreference: An instance of the OntologyAPISearchPreference class.
+            """
+            return OntologyAPISearchPreference(**data)
+
 class Coding:
-    def __init__(self, code, display="", system=None, description=""):
+    def __init__(self, code, display="", system=None, description="", api_preference=None):
```
Comment (torstees) [inline src/locutus/model/terminology.py:79]: I'm not really loving this. While this is a legitimate place to store these, I'm more inclined to see it as something that would live in it's own collection, similar to how we are storing mappings and provenance. The reason being that these codings are intended to represent something very basic, and these extensions are layers that are very specific to our application. If we stored information like this in the coding, we would need to filter it out before passing code systems and value sets to other systems (and the front end in certain places)

File: src/locutus/model/terminology.py [hunk 2]
```diff
@@ -93,6 +129,9 @@ class ChangeType(StrEnum):
         ApprovalRequested = "Approval Requested"
         Approved = "Approved"
         ApprovalDenied = "Approval Denied"
+        AddPreference = "Add API Ontology Preference"
```
Comment (torstees) [inline src/locutus/model/terminology.py:132]: I see now why you were adding that editor property. However, I don't think these are necessary.

**Review by torstees — CHANGES_REQUESTED** [review]
I think most of the work is fine, but there is a design issue I have with storing the preferences directly inside the coding. Also, there are some more minor issues mentioned in comments.

File: src/locutus/api/table.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 from flask import request
 from locutus import persistence
 from locutus.model.table import Table as mTable
+from locutus.model.terminology import Terminology
```
Comment (brendagutman) [inline src/locutus/api/table.py:5]: required for the `ChangeType` ln ~162

File: README.md [hunk 4]
```diff
@@ -1333,6 +1430,104 @@ If the root object in the body is missing both the "variable" and the
...
+#### DELETE
+
+Delete the api search preferences for a specific table (with a given id). An example of the data after removal shown below.
+```json
+{}
+```
+
+### https://[APPURL]/api/Table/[id]/filter/[code]
+
+<h4 id="table_id_filter_code_get">GET</h4>
```
Comment (torstees) [inline README.md:1479]: Those "#" characters are this flavor of Markdown's header indication. So, H4 is four pound signs: "#### GET" would be the same as your h4 tag above, unless you need that id to be specific.

**Review by torstees — APPROVED** [review]
This looks great! I did suggest replacing your H4 tags with the markdown equivalent of "####" in your README, if you don't need to specify that id yourself just to be consistent.

**Pull request merged**